src/Nox.Generator/Application/Factories/EntityFactory.template.cs

```diff
@@ -53,11 +53,19 @@ public abstract class {{className}}Base: IEntityFactory<{{entity.Name}},{{entity
     {
         var entity = new {{codeGeneratorState.DomainNameSpace}}.{{entity.Name}}();
         {{- for key in entity.Keys }}
-            {{- if key.Type == "Nuid" || key.Type == "AutoNumber" || key.Type == "DatabaseGuid" -}}
+            {{- if key.Type == "Nuid" || key.Type == "AutoNumber" || key.Type == "Guid" -}}
                 {{ continue; -}}
             {{- end }}
         entity.{{key.Name}} = {{ entity.Name }}.Create{{key.Name}}(createDto.{{key.Name}});
         {{- end }}
+        {{- for key in entity.Keys ~}}
+            {{- if key.Type == "Nuid" }}
+        entity.Ensure{{key.Name}}();
+            {{- end }}
+            {{- if key.Type == "Guid" }}
+        entity.{{key.Name}} = Nox.Types.Guid.From(System.Guid.NewGuid());    
```

Comment:
We should check if the CreateDto.{{key.Name}} is not provided, then to set new Guid for the entity.{{key.Name}}

Reply:
Desired changes were applied